src/keys.rs

```diff
@@ -281,76 +292,60 @@ impl IssuanceAuthorizingKey {
             .map(|esk| esk.sk().into())
     }
 
-    /// Derives the RedPallas signing key from isk. Internal use only, does not enforce all constraints.
-    fn derive_inner(&self) -> pallas::Scalar {
-        to_scalar(PrfExpand::ZsaIsk.expand(&self.0))
-    }
-
     /// Sign the provided message using the `IssuanceAuthorizingKey`.
-    pub fn sign(
-        &self,
-        rng: &mut (impl RngCore + CryptoRng),
-        msg: &[u8],
-    ) -> redpallas::Signature<IssuanceAuth> {
-        conditionally_negate(self.derive_inner()).sign(rng, msg)
+    pub fn sign(&self, msg: &[u8]) -> schnorr::Signature {
+        let schnorr_sk = schnorr::SigningKey::from_bytes(&self.0).unwrap();
+        schnorr_sk.sign(msg)
     }
 }
 
 /// A key used to validate issuance authorization signatures.
 ///
 /// Defined in [ZIP 227: Issuance of Zcash Shielded Assets § Issuance Key Generation][IssuanceZSA].
-/// Note that this is $\mathsf{ik}^\mathbb{P}$, which by construction is equivalent to
-/// $\mathsf{ik}$ but stored here as a RedPallas verification key.
 ///
 /// [IssuanceZSA]: https://qed-it.github.io/zips/zip-0227#issuance-key-derivation
-#[derive(Debug, Clone, PartialOrd, Ord)]
-pub struct IssuanceValidatingKey(VerificationKey<IssuanceAuth>);
+#[derive(Debug, Clone)]
+pub struct IssuanceValidatingKey(schnorr::VerifyingKey);
 
 impl From<&IssuanceAuthorizingKey> for IssuanceValidatingKey {
     fn from(isk: &IssuanceAuthorizingKey) -> Self {
-        IssuanceValidatingKey((&(conditionally_negate(isk.derive_inner()))).into())
-    }
-}
-
-impl From<&IssuanceValidatingKey> for pallas::Point {
-    fn from(issuance_validating_key: &IssuanceValidatingKey) -> pallas::Point {
-        pallas::Point::from_bytes(&(&issuance_validating_key.0).into()).unwrap()
+        let schnorr_sk = schnorr::SigningKey::from_bytes(&isk.0).unwrap();
+        IssuanceValidatingKey(*schnorr_sk.verifying_key())
     }
 }
 
 impl PartialEq for IssuanceValidatingKey {
     fn eq(&self, other: &Self) -> bool {
-        <[u8; 32]>::from(&self.0).eq(&<[u8; 32]>::from(&other.0))
+        self.to_bytes().eq(&other.to_bytes())
     }
 }
 
 impl Eq for IssuanceValidatingKey {}
 
 impl IssuanceValidatingKey {
-    /// Converts this spend validating key to its serialized form,
-    /// I2LEOSP_256(ik).
+    /// Converts this issuance validating key to its serialized form,
+    /// in big-endian order as defined in BIP 340.
     pub fn to_bytes(&self) -> [u8; 32] {
-        // This is correct because the wrapped point must have ỹ = 0, and
-        // so the point repr is the same as I2LEOSP of its x-coordinate.
-        <[u8; 32]>::from(&self.0)
+        // This provides the big-endian encoding, via the FieldBytesEncoding trait in the k256 crate.
+        U256::decode_field_bytes(&self.0.to_bytes()).to_be_bytes()
```

Comment:
- why not simply `self.0.to_bytes().into()` ?
- We need a round trip test for this function

Reply:
Done